random scripts-module/livescripts/randomenchants.ts

```diff
@@ -0,0 +1,61 @@
+@CharactersTable
+export class playerEnchantInfo extends DBEntry {
+    @DBPrimaryKey
+    playerGUID: uint64 = 0
+    @DBPrimaryKey
+    itemGUID: uint64 = 0
+    @DBField
+    display: uint32 = 0
+
+    constructor(player: TSPlayer, itemGUID: TSItem) {
+        super();
+        this.playerGUID = player.GetGUID()
+        this.itemGUID = itemGUID.GetGUID()
+    }
+
+    static get(player: TSPlayer, item: TSItem): playerEnchantInfo {
+        return player.GetObject('playerEnchantInfo', LoadDBEntry(new playerEnchantInfo(player, item)))
+    }
+}
+
+let enchants = [3789, 3854, 3273, 3225, 3870, 1899, 2674, 2675, 2671, 2672, 3365, 2673, 2343, 425, 3855, 1894, 1103, 1898, 3345, 1743, 3093, 1900, 3846, 1606, 283, 1, 3265, 2, 3, 3266, 1903, 13, 26, 7, 803, 1896, 2666, 25]
+let enchantsLength = enchants.length
+let chance = 1//0.25
+
+export function itemRandomEnchantVisual(events: TSEvents) {
+    events.Player.OnLogin((player, first) => {
+        applyVisuals(player)

```

Comment:
We may want to restrict the scope of this a bit in the future, but shouldn't be hard to do/should be okay for now.